### PR TITLE
docs: remove unnecessary steps and update examples

### PR DESCRIPTION
**Pull request by bgdanix**
Updated documentation with the following:

- remove the step of getting the `WORKLOAD_IDENTITY_PROVIDER_ID` as it's not being needed and used anywhere
- update `Usage` and `Examples` to make it compliant with GitHub Actions so that someone can copy-paste the example code, without getting complains from actions validity

### CODE REVIEW

**Comment by bgdanix** [comment]
@googlebot I signed it!

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ and permissions on Google Cloud.
 
 ```yaml
 jobs:
-  run:
+  job_id:
```
Comment (sethvargo) [inline README.md:33]: Can you tell me more about this change? Why is this necessary and what does it achieve?

Comment (bgdanix) [reply README.md:33]: If someone was to copy-paste that code into a GitHub action, then the following error message would be thrown: "A job requires either runs-on or uses"

**Review by sethvargo — APPROVED** [review]

**Comment by bgdanix** [comment]
I guess it's failing because I don't have contributor access?

**Comment by sethvargo** [comment]
Yea, that's a known issue. @bharathkkb or @averikitsch I re-ran the tests as myself, but it still used the original submitters perms. Let me know if you how you want to merge this.

**Comment by bharathkkb** [comment]
@sethvargo I believe the rerun only works for dependabot PRs. Since this is a readme change I think we can merge it as is, otherwise for code changes I usually create a branch off main merge contributor PR into that branch and reopen a PR with that branch.
